crates/sparrow-runtime/src/prepare.rs

```diff
@@ -409,4 +413,117 @@ mod tests {
         let _prepared_schema = prepared_batch.schema();
         let _metadata_schema = metadata.schema();
     }
+
+    #[tokio::test]
+    async fn test_preparation_single_entity_key_slicing() {
```

Comment:
Some debugging involved looking at different stacks. Missed some tests in preparation so added some new ones.